File: AU/AU/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace AU.Models
{
    public class Room
    {
        public int RoomId { get; set; }
        public int BuildId { get; set; }
        public Build? Build { get; set; }
        public string? RoomName { get; set; }
        public int RoomTypeId { get; set; }
        public RoomType? RoomType { get; set; }
        public string Capacity { get; set; }
        public string Floor { get; set; }
        public string RoomNumber { get; set; }
    }
}

File: AU.UI/AU.UI/src/app/models/room.ts
import { Build } from "./build";
import { RoomType } from "./roomType";

export class Room {
    roomId: number;
    buildId: number;
    build?: Build;
    roomName?: string;
    roomTypeId: number;
    roomType?: RoomType;
    capacity: string;
    floor: string;
    roomNumber: string;
}